plugins/awards-backend/src/service/router.ts

```diff
@@ -154,6 +167,42 @@ export async function createRouter(
     response.json(award);
   });
 
+  router.post('/logos/upload', async (request, response) => {
+    await getUserRef(identity, request);
+
+    if (!request.files) {
+      response.status(400).send('No files were uploaded.');
+      return;
+    }
+    const logo = request.files.file;
+    if (Array.isArray(logo)) {
+      response.status(400).send('Must only upload one file.');
+      return;
+    }
+
+    const key = await awardsApp.uploadLogo(logo.data, logo.mimetype);
+
+    const apiUrl = await options.discovery.getExternalBaseUrl('awards');
+    response.status(201).json({
+      location: `${apiUrl}/logos/${key}`,
+    });
+  });
+
+  router.get('/logos/:key', async (request, response) => {
```

Comment:
so here we are essentially creating an s3 proxy so we can serve images from a private s3 bucket.

my main q is: do we want to just use a public s3 bucket here? only because it’d be nice to post award logos in automated slack messages. we may also be able to use presigned urls for sharing images from a private bucket to slack though not 100% sure